src/main/java/hudson/plugins/validating_string_parameter/ValidatingStringParameterDefinition.java

```diff
@@ -118,6 +119,11 @@ public FormValidation doCheckRegex(@QueryParameter final String value) {
 
         /**
          * Called to validate the passed user entered value against the configured regular expression.
+         *
+         * @param regex the regex
+         * @param failedValidationMessage message to display when validation fails
+         * @param value the value to be validated
+         * @return whether the value is valid according to the specified regex
          */
         public FormValidation doValidate(@QueryParameter("regex") String regex,

```

Comment:
Nice to have `@Restricted(NoExternalUse.class)`. It prevents usage of Stapler methods in dependent plugins